src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.h

```diff
@@ -0,0 +1,171 @@
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId The ID of the operational state derived cluster to be instantiated.
+     */
+    Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId);
+
+    ~Instance() override;
+
+    /**
+     * @brief Initialise the Microwave Oven Control server instance.
+     * This function must be called after defining an Instance class object.
+     * @return Returns an error if the given endpoint and cluster ID have not been enabled in zap or if the
+     * CommandHandler or AttributeHandler registration fails, else returns CHIP_NO_ERROR.
+     */
+    CHIP_ERROR Init();
+
+    /**
+     * @brief defined the get/set api for the mandatory attributes
+     */
+    uint32_t GetCookTime() const;
+    void SetCookTime(uint32_t cookTime);
+
+    uint8_t GetPowerSetting() const;
+    void SetPowerSetting(uint8_t powerSetting);
+
+private:
+    Delegate * mDelegate;
+    EndpointId mEndpointId;
+    ClusterId mClusterId;
+
+    /**
+     * set default values
+     */
+    uint32_t mCookTime   = kDefaultCookTime;
+    uint8_t mPowerSettng = kDefaultPowerSetting;
+
+    /**
+     * IM-level implementation of read
+     * @return appropriately mapped CHIP_ERROR if applicable (may return CHIP_IM_GLOBAL_STATUS errors)
+     */
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    /**
+     * @brief Inherited from CommandHandlerInterface
+     */
+    void InvokeCommand(HandlerContext & ctx) override;
+
+    /**
+     * @brief Handle Command: SetCookingParameters.
+     */
+    void HandleSetCookingParameters(HandlerContext & ctx, const Commands::SetCookingParameters::DecodableType & req);
+
+    /**
+     * @brief Handle Command: AddMoreTime.
+     */
+    void HandleAddMoreTime(HandlerContext & ctx, const Commands::AddMoreTime::DecodableType & req);
+};
+
+/** @brief
+ *  Defines methods for implementing application-specific logic for the MicrowaveOvenControl Cluster.
+ */
+class Delegate
+{
+public:
+    Delegate() = default;
+
+    virtual ~Delegate() = default;
+
+    /**
+     *   @brief Handle Command Callback in application: SetCookingParameters
+     *   @return Returns the Interaction Model status code which was user determined in the business logic
+     */
+    virtual Protocols::InteractionModel::Status
+    HandleSetCookingParametersCallback(Optional<uint8_t> cookMode, Optional<uint32_t> cookTime, Optional<uint8_t> powerSetting) = 0;
+
+    /**
+     *   @brief Handle Command Callback in application: AddMoreTime
+     *   @return Returns the Interaction Model status code which was user determined in the business logic
+     */
+    virtual Protocols::InteractionModel::Status HandleAddMoreTimeCallback(uint32_t timeToAdd) = 0;
+
+    /**
+     *   @brief defined the get/set interface for MinPower
```

Comment:
Please fix the grammar, but also: the setters for these attributes are completely unused, no?  Why are they here?  These attributes are compile-time constants, per spec.

Reply:
Why the setters are here is because we want to let application to determine the value of attributes , from the spec that "**For example, if MinPower is 1, MaxPower is 10, and PowerSetting can be set to any integer between MinPower and MaxPower, PowerStep would be set to 1**", we should reserve the setters for application user to set the value. Otherwise, the 3 attributes are assgined default value.

Reply:
But the _SDK is not going to call those setters_, right?  Then why are they part of the API contract between the SDK and the application?

Reply:
got your mean, I will remove the setters. The application can set the values by itself.